src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefReturnTests.cs

```diff
@@ -1672,5 +1672,536 @@ .maxstack  1
 }");
         }
 
+        [Fact]
+        public void RefReturnConditionalAccess01()
+        {
+            var text = @"
+    using System;
+ 
+    class Program
+    {
+        class C1<T> where T : IDisposable
+        {
+            T inst = default(T);
+
+            public ref T GetDisposable()
+            {
+                return ref inst;
+            }
+
+            public void Test()
+            {
+                GetDisposable().Dispose();
+                System.Console.Write(inst.ToString());
+
+                GetDisposable()?.Dispose();
```

Comment:
I am not sure other cases are interesting. The case we have here requires special codegen since we have T for which we do not know statically if it is a class or a struct. 
In other cases, the fact that receiver is byref would not introduce anything special.